apps/web/modules/ui/components/id-badge/index.test.tsx

```
@@ -0,0 +1,139 @@
+    const copyButton = container.querySelector('button[aria-label="Copy HOVER-ID"]');
+    expect(copyButton).toBeInTheDocument();
+
+    // Mouse leave - button should disappear
+    fireEvent.mouseLeave(badge);
+    expect(container.querySelector("button")).toBeNull();
+  });
+
+  test("renders copy button with correct accessibility attributes", () => {
+    const { container } = render(<IdBadge id="TEST-001" />);
+
+    const copyButton = container.querySelector('button[aria-label="Copy TEST-001"]')!;
```

Biome (1.9.4)

[error] Forbidden non-null assertion.



(lint/style/noNonNullAssertion)